### PR TITLE
Scavengers: Add Jump Start

### PR DESCRIPTION
**Pull request by sparkychildcharlie**
_Nyuuu_ game mode (last of the game modes for May!)

This is approved by Meicoo-sama!  よろしくお願いし _”にゃ”_ す！
image

### CODE REVIEW

**Comment by Zarel** [comment]
Do you think you can follow the TS rules (line limit 120, consistent quotes)?

File: server/chat-plugins/scavenger-games.js [hunk 1]
```diff
@@ -253,15 +252,15 @@ const MODES = {
 		round: 0,
 		playerlist: null,
 	},
-
-	'ko': 'kogames',
-	kogames: {
-		name: 'KO Games',
-		id: 'kogames',
+	'scav': 'scavengergames',
+	'scavgames': 'scavengergames',
```
Comment (Zarel) [inline server/chat-plugins/scavenger-games.js:256]:
```suggestion
	scav: 'scavengergames',
	scavgames: 'scavengergames',
```

File: server/chat-plugins/scavenger-games.js [hunk 2]
```diff
@@ -186,15 +186,14 @@ const TWISTS = {
 };
 
 const MODES = {
-	'scav': 'scavengergames',
-	'scavgames': 'scavengergames',
-	scavengergames: {
-		name: 'Scavenger Games',
-		id: 'scavengergames',
+	'ko': 'kogames',
```
Comment (Zarel) [inline server/chat-plugins/scavenger-games.js:189]:
```suggestion
	ko: 'kogames',
```

File: server/chat-plugins/scavenger-games.js [hunk 3]
```diff
@@ -344,12 +343,91 @@ const MODES = {
...
+					return true;
+				}
+			},
+
+			onViewHunt(user) {
+				if (this.answerLock && !(this.hosts.some(h => h.id === user.id) || user.id === this.staffHostId)) return true;
+			},
+
+			onCreateCallback() {
+				if (this.answerLock) return `|raw|<div class="broadcast-blue"><strong>${['official', 'unrated'].includes(this.gameType) ? 'An' : 'A'} ${this.gameType} Scavenger Hunt by <em>${Chat.escapeHTML(Chat.toListString(this.hosts.map(h => h.name)))}</em> has been started${(this.hosts.some(h => h.id === this.staffHostId) ? '' : ` by <em>${Chat.escapeHTML(this.staffHostName)}</em>`)}.<br />The first hint is currently being handed out to early finishers.`;
```
Comment (Zarel) [inline server/chat-plugins/scavenger-games.js:404]:
```suggestion
				if (this.answerLock) {
					return `|raw|<div class="broadcast-blue"><strong>${['official', 'unrated'].includes(this.gameType) ? 'An' : 'A'} ${this.gameType} Scavenger Hunt by <em>${Chat.escapeHTML(Chat.toListString(this.hosts.map(h => h.name)))}</em> has been started${(this.hosts.some(h => h.id === this.staffHostId) ? '' : ` by <em>${Chat.escapeHTML(this.staffHostName)}</em>`)}.<br />The first hint is currently being handed out to early finishers.`;
				}
```

**Comment by Zarel** [comment]
Ideally also think about a `.ts` conversion for real, but I won't ask you to do it here.

**Comment by sparkychildcharlie** [comment]
> Ideally also think about a `.ts` conversion for real, but I won't ask you to do it here.

I can consider doing it, but TypeScript in general is "exploring new territory for me"

> Do you think you can follow the TS rules (line limit 120, consistent quotes)?

ooops i never knew about that

File: server/chat-plugins/scavenger-games.js [hunk 4]
```diff
@@ -344,12 +344,96 @@ const MODES = {
...
+
+			onJoin(user) {
+				if (this.answerLock) {
+					user.sendTo(this.room, `The hunt is not open for guesses yet!`);
+					return true;
+				}
+			},
+
+			onViewHunt(user) {
+				if (this.answerLock && !(this.hosts.some(h => h.id === user.id) || user.id === this.staffHostId)) return true;
```
Comment (Zarel) [inline server/chat-plugins/scavenger-games.js:401]:
```suggestion
				if (this.answerLock && !(this.hosts.some(h => h.id === user.id) || user.id === this.staffHostId)) {
					return true;
				}
```

**Pull request merged**